PythonRpcServer/scenedetector.py

```diff
@@ -1,113 +1,124 @@
 import os
-import cv2
+from cv2 import cv2
 import json
-from time import perf_counter 
-
-
+import numpy as np
+from skimage.metrics import structural_similarity as ssim
+from datetime import datetime
 
 DATA_DIR = os.getenv('DATA_DIRECTORY')
 
 
-# Standard PySceneDetect imports:
-from scenedetect.video_manager import VideoManager
-from scenedetect.scene_manager import SceneManager
-# For caching detection metrics and saving/loading to a stats file
-from scenedetect.stats_manager import StatsManager
+def find_scenes(video_path, min_scene_length=1, abs_min=0.75, abs_max=0.98, find_subscenes=True, max_subscenes_per_minute=12):
+    """
+    Detects scenes within a video. 
+    
+    Calculates the structual similarity index measure (SSIM) between each subsequent frame then uses
+    the list of SSIMs to identify where scene changes are. 
+    
+    Parameters:
+    video_path (string): Path of the video to be used.
+    min_scene_length (int): Minimum scene length in seconds. Default 1s
+    abs_min (float): Minimum SSIM value for non-scene changes, i.e. any frame with SSIM < abs_min 
+        is defined as a scene change. Default 0.7
+    abs_max (float): Maximum SSIM value for scene_changes, i.e. any frame with SSIM > abs_max
+        is defined as NOT a scene change. Default 0.98
+    find_subscenes (boolean): Find subscenes or not. Default True
+    max_subscenes_per_minute (int): Maximum number of subscenes per minute within a scene. If number
+        of subscenes found exceeds max_subscenes_per_minute, then none of those subscenes are returned.
+        Rational is that too many detected subscenes is more likely a result of a video clip or other
+        noisy media and not actual scene changes. 
+    
+    Returns:
+    string: List of dictionaries dumped to a JSON string. Each dict corresponds to a scene/subscene,
+        with the key/item pairs being starting timestamp (start), image file name (img_file), ending 
+        timestamp (end), and boolean indicating if it's a scene or subscene (is_subscene).
+    """
+    
+    try:
 
-# For content-aware scene detection:
-from scenedetect.detectors.content_detector import ContentDetector
-from scenedetect.detectors.threshold_detector import ThresholdDetector
+        file_name = video_path[video_path.rfind('/')+1 : video_path.find('.')]
```

Comment:
python has  functions on os.path to extract the file_name which will be more robust than this